integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/handler/HandlerTest.java

```diff
@@ -98,6 +102,35 @@ public void testSynapseHandlerExecution()
 
     }
 
+    @Test(groups = { "wso2.esb" },
+            description = "Sending a message via proxy to check whether Synapse Handlers get invoked when "
+                    + " a SoapFault come as a response")
+    public void testSynapseHandlerExecutionWhenSoapFaultRecieved() throws IOException, LogViewerLogViewerException {
+        boolean responseInStatus = false;
+        boolean errorOnSoapFaultStatus = false;
+        logViewerClient.clearLogs();
+        try {
+            axis2Client.sendSimpleStockQuoteRequest(getProxyServiceURLHttp("handlerTestProxyWithSoapfault"), null,
+                    "WSO2");
+            fail("This query must throw an exception since SoapFault come as response");
```

Comment:
Shall we use expectedExceptions here to assert the exception

Reply:
The purpose of the test is to verify whether the synapse handlers get invoked when a soap fault is received from the back-end. 
So it has been asserted by errorOnSoapFaultStatus (see whether the fault sequence got invoked) and responseInStatus (see whether the handler gets invoked )properties. So I think there is no need for asserting the exception here.

Reply:
What I meant is to use the syntax expectedExceptions and we can have the expected message there also instead of fail here.
https://github.com/wso2/product-ei/blob/master/integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/endpoint/test/FailoverEndpointsForSingleLeafEndpointTestCase.java#L40

Reply:
Can't we use the syntax expectedExceptions instead of fail
https://github.com/wso2/product-ei/blob/master/integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/endpoint/test/FailoverEndpointsForSingleLeafEndpointTestCase.java#L40